pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -59,19 +107,32 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 		return emptyStat, fmt.Errorf("GET request for URL %q returned HTTP status %v", url, resp.StatusCode)
 	}
 	if resp.Header.Get("Content-Type") == network.ProtoAcceptContent {
-		return statFromProto(resp.Body)
+		return statFromProto(resp.Body, resp.ContentLength)
```

Comment:
`resp.ContentLength` is set by the http client when reading the resp, no need to do that at the header side at the protobustat report.

Reply:
Ok did that to be on the safe size. I thought it did so, anyway probably that is why the tests e2e fail.